player/src/lib.rs

```diff
@@ -0,0 +1,64 @@
+extern crate ipc_channel;
+#[macro_use]
+extern crate serde_derive;
+
+pub mod frame;
+pub mod metadata;
+
+use ipc_channel::ipc;
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub enum PlaybackState {
+    Stopped,
+    // Buffering,
+    Paused,
+    Playing,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub enum PlayerEvent {
+    EndOfStream,
+    MetadataUpdated(metadata::Metadata),
+    StateChanged(PlaybackState),
+    FrameUpdated,
+    Error,
+}
+
+pub trait Player: Send {
+    fn register_event_handler(&self, sender: ipc::IpcSender<PlayerEvent>);
+    fn register_frame_renderer(&self, renderer: Box<frame::FrameRenderer>);
+
+    fn setup(&self) -> bool;
+    fn play(&self);
+    fn stop(&self);
+
+    fn set_input_size(&self, size: u64);
+    fn push_data(&self, data: Vec<u8>) -> bool;
+    fn end_of_stream(&self) -> bool;
```

Comment:
`Result<(), ()>` as the return value of `setup`, `push_data` and `end_of_stream` is probably more idiomatic and let us extend it with specific errors in the future if needed.